task/task.go

```diff
@@ -66,6 +67,15 @@ func New(binary string, workingDir string, env map[string]string, args ...string
 	return &t
 }
 
+// String implements Stringer for Task, which is required for not tripping the
```

Comment:
Where is the race detected exactly? TBH locking in the String() method feels like a hack. Any way to fix the Task implementation instead to not trigger the race detector?

Reply:
It was in the mocks for the `TaskManager`. Whenever the mocking framework tried to format am error message using a Task instance it tripped the race detector; I'm assuming because the default implementation was non-atomically reading several fields in order to generate the string.

The tests that needed this via the mocking framework have now been deleted, so this has gone as well.